plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/PluginUtil.kt

```diff
@@ -37,12 +37,23 @@ internal inline fun <reified T : Task> Project.registerTask(
 
 internal const val EDITOR_CONFIG_FILE_NAME = ".editorconfig"
 
-internal fun getEditorConfigFiles(currentProject: Project): FileCollection {
-    return searchEditorConfigFiles(
+internal fun getEditorConfigFiles(
+    currentProject: Project,
+    additionalEditorconfigFile: RegularFileProperty
+): FileCollection {
+    var editorConfigFileCollection = searchEditorConfigFiles(
```

Comment:
Should we even search if the user has specified their own? If there is both one in the repository and one passed via the command line, which one does ktlint use?

Reply:
According to ktlint code - it is still using working directory `.editorconfig` file even if `--editorconfig` option was used:
- loading editorconfig passed via `--editorconfig` option:
https://github.com/pinterest/ktlint/blob/af7202536fd76cce9b3bb6ff0f81855de5060f3e/ktlint/src/main/kotlin/com/pinterest/ktlint/Main.kt#L379
- loading current working directory editorconfigs:
https://github.com/pinterest/ktlint/blob/af7202536fd76cce9b3bb6ff0f81855de5060f3e/ktlint/src/main/kotlin/com/pinterest/ktlint/Main.kt#L387

`editorConfigFileCollection` is just used as a task input, but never passed to the ktlint itself, so merging here all editorconfig files is right behaviour.

Though, actually, I've noticed that in this PR additional `.editorconfig` file was never passed to ktlint itself :+1:

Reply:
Fixed in https://github.com/JLLeitschuh/ktlint-gradle/pull/247/commits/eab563de0cfa9d561b32e00d77a7f427d463f903

Reply:
If it's an override, then it should be called an override, not `additional`. `additional` sounds like they are being merged or something.

Reply:
Sorry, maybe I was not clear - ktlint adds additional `.editorconfig` file to the `.editorconfig` files that are detected in the working directory.

Reply:
I think I understand now. I didn't see that it's an `add` below. 🤦‍♂